pkg/controller/nodes/task/k8s/plugin_collector.go

```diff
@@ -0,0 +1,113 @@
+package k8s
+
+import (
+	"context"
+	"github.com/lyft/flytestdlib/logger"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	"runtime/pprof"
+	"strings"
+	"time"
+
+	"github.com/lyft/flytestdlib/contextutils"
+	"github.com/lyft/flytestdlib/promutils"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const resourceLevelMonitorCycleDuration = 10 * time.Second
+const KindKey contextutils.Key = "kind"
+
+// This object is responsible for emitting metrics that show the current number of Flyte workflows, cut by project and domain.
+// It needs to be kicked off. The periodicity is not currently configurable because it seems unnecessary. It will also
+// a timer measuring how long it takes to run each measurement cycle.
+type ResourceLevelMonitor struct {
+	Scope promutils.Scope
+
+	// Meta timer - this times each collection cycle to measure how long it takes to collect the levels GaugeVec below
+	CollectorTimer *promutils.StopWatchVec
+
+	// System Observability: This is a labeled gauge that emits the current number of FlyteWorkflow objects in the informer. It is used
+	// to monitor current levels. It currently only splits by project/domain, not workflow status.
+	levels *prometheus.GaugeVec
+
+	// This informer will be used to get a list of the underlying objects that we want a tally of
+	sharedInformer cache.SharedIndexInformer
+
+	// The kind here will be used to differentiate all the metrics, we'll leave out group and version for now
+	gvk schema.GroupVersionKind
+}
+
+func (r *ResourceLevelMonitor) countList(ctx context.Context, objects []interface{}) map[string]int {
+	// Map of namespace to counts
+	counts := map[string]int{}
+
+	// Collect all workflow metrics
+	for _, v := range objects {
+		metadata, err := meta.Accessor(v)
+		if err != nil {
+			logger.Errorf(ctx, "Error converting obj %v to an Accessor %s\n", v, err)
+			continue
+		}
+		counts[metadata.GetNamespace()]++
```

Comment:
Worth noting that in L5, everything will look like it was launched in the same namespace because they do not use namespaces to differentiate projects... I _think_ we add project and domain to the labels?

Reply:
but does the spark operator add labels to its pods?